src/id3/v2/frame/read.rs

```diff
@@ -11,23 +11,43 @@ use std::io::Read;
 
 use byteorder::{BigEndian, ReadBytesExt};
 
-impl<'a> Frame<'a> {
+pub(crate) enum ParsedFrame<'a> {
+	Next(Frame<'a>),
+	Skipped,
+	Eof,
+}
+
+impl<'a> ParsedFrame<'a> {
 	pub(crate) fn read<R>(
 		reader: &mut R,
 		version: Id3v2Version,
 		parse_mode: ParsingMode,
-	) -> Result<(Option<Self>, bool)>
+	) -> Result<Self>
 	where
 		R: Read,
 	{
 		// The header will be upgraded to ID3v2.4 past this point, so they can all be treated the same
-		let (id, mut size, mut flags) = match match version {
-			Id3v2Version::V2 => parse_v2_header(reader)?,
-			Id3v2Version::V3 => parse_header(reader, false)?,
-			Id3v2Version::V4 => parse_header(reader, true)?,
-		} {
-			None => return Ok((None, true)),
-			Some(frame_header) => frame_header,
+		let parse_header_result = match version {
+			Id3v2Version::V2 => parse_v2_header(reader),
+			Id3v2Version::V3 => parse_header(reader, false),
+			Id3v2Version::V4 => parse_header(reader, true),
+		};
+		let (id, mut size, mut flags) = match parse_header_result {
+			Ok(None) => {
+				// Stop reading
+				return Ok(Self::Eof);
+			},
+			Ok(Some(some)) => some,
+			Err(err) => {
+				match parse_mode {
+					ParsingMode::Strict => return Err(err),
+					ParsingMode::BestAttempt | ParsingMode::Relaxed => {
+						// Skip this frame and continue reading
```

Comment:
I added a test for this, and just realized that the frame will not actually be skipped over.

There would need to be a way to get the `size` out of the header even if there's an error.

Reply:
Fixed.

Code could be improved later to avoid the mutable reference param. Not urgent.

Reply:
The naming of `ParsedFrame` is more appropriate now, because the skipping actually happens later.